lib/bridge/MatrixHandler.js

```diff
@@ -1262,6 +1262,33 @@ MatrixHandler.prototype._sendIrcAction = Promise.coroutine(
     }
 
     let text = event.content.body;
+    const REPLY_SOURCE_MAX_LENGTH = 32;
+    const REPLY_NAME_MAX_LENGTH = 12;
+    if (event.content["m.relates_to"] && event.content["m.relates_to"]["m.in_reply_to"]) {
+        // This is a reply.
+        const match = /> <(@.*:.*)>(.*)\n\n(.*)/.exec(text);
+        if (match.length === 4) {
+            let rplName;
+            let sourceLength = REPLY_SOURCE_MAX_LENGTH - "...".length;
+            const rplSource = match[2].substr(0, sourceLength) +
+                              (match[2].length > sourceLength ? "..." : "");
+            const rplText = match[3];
+
+            // Fetch the sender's IRC nick.
+            const sourceClient = this.ircBridge.getIrcUserFromCache(ircRoom.server, match[1]);
+            if (sourceClient) {
+                rplName = sourceClient.nick;
+            }
+            else {
+                // Somehow we failed, so fallback to userid.
+                rplName = match[1].substr(1,
+                    Math.min(REPLY_NAME_MAX_LENGTH, userid.indexOf(":") - 1)
+                );
+            }
+            text = `<${rplName} "${rplSource.trim()}"> ${rplText}`;
```

Comment:
If we're going to trim, surely we should trim ahead of getting to this point?

Reply:
Point